### PR TITLE
Zenml weak credentials

### PR DESCRIPTION
**Pull request by secureness**
PRP: https://github.com/google/tsunami-security-scanner-plugins/issues/444
testbeds: https://github.com/google/security-testbeds/pull/57

### CODE REVIEW

**Review by leonardo-doyensec — COMMENTED** [review]
Hello @secureness.
Thank you for your contribution. There are several issues to address. You can find all of them in the comments.

Feel free to reach out.
~ Leonardo (Doyensec)

File: google/detectors/credentials/generic_weak_credential_detector/src/main/resources/detectors/credentials/genericweakcredentialdetector/data/service_default_credentials.textproto [hunk 1]
```diff
@@ -77,3 +77,9 @@ service_default_credentials {
   default_usernames: "username"
   default_passwords: "password"
 }
+
+service_default_credentials {
+  service_name: "zenml"
+  default_usernames: "default"
+  default_passwords: ""
+}
```
[the remark this answers is not in the document]
Comment (leonardo-doyensec) [reply google/detectors/credentials/generic_weak_credential_detector/src/main/resources/detectors/credentials/genericweakcredentialdetector/data/service_default_credentials.textproto:85]: Perfect. This was not clear at the beginning. After running Tsunami with the fingerprint plugin, everything works fine on my side.

File: google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/zenml/ZenMlCredentialTester.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    // return empty credential instead of all the weak credentials
+    return credentials.stream()
+        .filter(cred -> isZenMlAccessible(networkService, cred))
+        .findFirst()
+        .map(ImmutableList::of)
+        .orElseGet(ImmutableList::of);
+  }
+
+  private boolean isZenMlAccessible(NetworkService networkService, TestCredential credential) {
+    logger.atWarning().log(
```
Comment (pisqu4red) [inline google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/zenml/ZenMlCredentialTester.java:86]: We have test breaking on our side:

error: [FloggerLogString] Arguments to log(String) must be compile-time constants or parameters annotated with @CompileTimeConstant. If possible, use Flogger's formatting log methods instead.

**Review by pisqu4red — CHANGES_REQUESTED** [review]

**Comment by tooryx** [comment]
Hey @secureness, I will check with the rest of the team.

**Pull request merged**